components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -932,20 +1170,26 @@ private static void addAPIPipeline(Connection connection, List<Pipeline> pipelin
                 }
                 statement.setLong(5, pipeline.getQuotaPolicy().getLimit().getUnitTime());
                 statement.setString(6, pipeline.getQuotaPolicy().getLimit().getTimeUnit());
-                statement.execute();
-                List<Integer> conditionGroupIds = getConditionGroupIDs(connection, uuid);
-                int conID = conditionGroupIds.get(conditionGroupIds.size() - 1); //getting the last condition group id
-                List<Condition> conditionList = pipeline.getConditions();
-                for (Condition condition : conditionList) {
-                    if (condition instanceof IPCondition) {
-                        addIPCondition(connection, (IPCondition) condition, conID);
-                    } else if (condition instanceof HeaderCondition) {
-                        addHeaderCondition(connection, (HeaderCondition) condition, conID);
-                    } else if (condition instanceof JWTClaimsCondition) {
-                        addJWTClaimCondition(connection, (JWTClaimsCondition) condition, conID);
-                    } else if (condition instanceof QueryParameterCondition) {
-                        addParamCondition(connection, (QueryParameterCondition) condition, conID);
+                statement.executeUpdate();
+                ResultSet rs = statement.getGeneratedKeys();
+                if (rs.next()) {
+                    // get the auto increment id
+                    int conditionId = rs.getInt(1);
+                    List<Condition> conditionList = pipeline.getConditions();
+                    for (Condition condition : conditionList) {
+                        if (condition instanceof IPCondition) {
```

Comment:
Misuse of Object Oriented concepts, the fact that you need to use instanceOf to determine the actual type. This is a lazy design approach. This needs to be re-factored to work with the Condition type itself.

Worst thing about this kind of code is it teaches junior team members to do the wrong thing!